### PR TITLE
[VOL-137] Dvol config set key

### PR DESCRIPTION
**Pull request by robhaswell**
https://clusterhq.atlassian.net/browse/VOL-137

Implement setting a key with `dvol config`. Next up: Retrieving a key. I will create an integration test in the next branch.

**NOTE** This branch does not yet actually save a configuration. This will be added next.

This branch also makes some minor modifications to how error handling is performed.

### CODE REVIEW

File: pkg/cmd/config_test.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
+package cmd
+
+import (
+	"os"
+	"testing"
+
+	"github.com/spf13/viper"
+)
+
+func TestCannotGetYet(t *testing.T) { // TODO: Implement get and then remove this
```
Comment (robhaswell) [inline pkg/cmd/config_test.go]: I wanted good test coverage.

File: pkg/cmd/config.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
```
Comment (zubron) [inline pkg/cmd/config.go]: This needs to be added as a dependency.

File: pkg/cmd/config.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+		return true
+	}
+	return false
+}
+
+func NewCmdConfig(out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "config name [value]",
+		Short: "Get or set global options",
+		RunE: func(cmd *cobra.Command, args []string) error {
```
Comment (zubron) [inline pkg/cmd/config.go]: I'm not sure if it's `RunE` causing this, but when there is an error condition, the error string is printed twice.

Comment (robhaswell) [inline pkg/cmd/config.go]: Good spot, it means I need to convert everything to using `RunE` instead of the old ad-hoc error message implementation. Damn.

Comment (zubron) [inline pkg/cmd/config.go]: Good idea. The double error printing is coming from `dvol.go`. If `RootCmd.Execute()` fails, we print the error that is returned.

Comment (robhaswell) [inline pkg/cmd/config.go]: Yep - just waiting for a build on my laptop.

Comment (robhaswell) [inline pkg/cmd/config.go]: Nothing else was returning an error at any point so I could just remove that line in https://github.com/ClusterHQ/dvol/pull/63/commits/f8b3fe7892e6aa9d07753dfbd590847d0931168f and all the tests still pass.

**Comment by zubron** [comment]
Thanks @robhaswell! Looks good :)

**Pull request merged**